src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +225,296 @@ def execute(self, reqStatus):
+        return wflow
+
+    def getMSOutputTransferInfo(self, wflow):
+        """
+        Fetches the transfer information from the MSOutput REST interface for
+        the given workflow.
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :return wflow:
+        """
+        headers = {'Accept': 'application/json'}
+        params = {}
+        url = '%s/data/info?request=%s' % (self.msConfig['MSOutputUrl'],
+                                           wflow['RequestName'])
+        mgr = RequestHandler()
+        res = mgr.getdata(url, params=params, headers=headers, ckey=ckey(), cert=cert())
+        data = json.loads(res)['result'][0]
+        wflow['TransferInfo'] = data['transferDoc']
+        return wflow
+
+    def getRucioRules(self, wflow, gran, rucioAcct):
+        """
+        Queries Rucio and builds the relevant list of blocklevel rules for
+        the given workflow
+        :param  wflow:   A MSRuleCleaner workflow representation
+        :param  gran:    Data granularity to search for Rucio rules. Possible values:
+                        'block' || 'container'
+        :return:        wflow
+        """
+        currPline = wflow['PlineMarkers'][-1]
+        # Find all the output placement rules crated by the agents
+        # NOTE: Check MSOutput 'TransferStatus' first and assure it is 'done'
+        #       only then we proceed with building the respective list of rules
+        if wflow['TransferInfo'] is not None \
+           and wflow['TransferInfo']['TransferStatus'] == 'done': # FIXME 'pennding' -> 'done'
+            for dataCont in wflow['OutputDatasets']:
+                if gran == 'container':
+                    wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(dataCont, account=rucioAcct))
+                elif gran == 'block':
+                    blocks = self.rucio.getBlocksInContainer(dataCont)
+                    for block in blocks:
+                        wflow['RulesToClean'][currPline].extend(self.rucio.listDataRules(block, account=rucioAcct))
+        return wflow
+
+    def cleanRucioRules(self, wflow):
+        """
+        Cleans all the Rules present in the field 'RulesToClean' in the MSRuleCleaner
+        workflow representation. And fills the relevant Cleanup Status.
+        :param wflow:   A MSRuleCleaner workflow representation
+        :return:        wflow
+        """
+        # NOTE: The function should be called independently and sequentially from
+        #       The Input and the respective BlockLevel pipelines.
+
+        # NOTE: The current functional pipeline is always the last one in the PlineMarkers list
+        currPline = wflow['PlineMarkers'][-1]
+        delResutls = []
+        if wflow['TransferInfo'] is not None \
+           and wflow['TransferInfo']['TransferStatus'] == 'done': # FIXME 'pennding' -> 'done'
+            if self.isRealMode:
+                for rule in wflow['RulesToClean'][currPline]:
```

Comment:
How about something like:
```
successfulDeletion = True
for rule in wflow['RulesToClean'][currPline]:
   if fail to delete any rule:
      successfulDeletion = False
...
wflow['CleanupStatus'][currPline] = successfulDeletion
```
?

Reply:
Exactly the same logic is already implemented internally into the `rucio.deleteRule()` method. See [1]. No need to repeat here I believe. Will gain nothing. So I already count on the fact that I have no way to get anything else from inside that method but a `Bool` value i.e. it already suppresses any exception that my be thrown by the third parity Rucio client. 

[1]
https://github.com/dmwm/WMCore/blob/51e3e2f6975b92abbbd9bf7291cf7e8d8eabd2e3/src/python/WMCore/Services/Rucio/Rucio.py#L717-L724

Reply:
My comment was on how you make sure if you have successfully deleted all the rules or not (basically your logic with `delResult`). Just a different way to evaluate it though.